components/dashboard/src/settings/PersonalAccessTokensCreateView.tsx

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { PersonalAccessToken } from "@gitpod/public-api/lib/gitpod/experimental/v1/tokens_pb";
+import dayjs from "dayjs";
+import { useContext, useEffect, useState } from "react";
+import { Redirect, useHistory, useParams } from "react-router";
+import { Link } from "react-router-dom";
+import Alert from "../components/Alert";
+import CheckBox from "../components/CheckBox";
+import DateSelector from "../components/DateSelector";
+import { SpinnerOverlayLoader } from "../components/Loader";
+import { FeatureFlagContext } from "../contexts/FeatureFlagContext";
+import { personalAccessTokensService } from "../service/public-api";
+import { PageWithSettingsSubMenu } from "./PageWithSettingsSubMenu";
+import { AllPermissions, TokenAction, TokenExpirationDays, TokenInfo } from "./PersonalAccessTokens";
+import { settingsPathPersonalAccessTokens } from "./settings.routes";
+import ShowTokenModal from "./ShowTokenModal";
+import { Timestamp } from "@bufbuild/protobuf";
+import arrowDown from "../images/sort-arrow.svg";
+
+interface EditPATData {
+    name: string;
+    expirationDays: string;
+    expirationDate: Date;
+    scopes: Set<string>;
+}
+
+const personalAccessTokenNameRegex = /^[a-zA-Z0-9-_ ]{3,63}$/;
+
+function PersonalAccessTokenCreateView() {
+    const { enablePersonalAccessTokens } = useContext(FeatureFlagContext);
+
+    const params = useParams<{ tokenId?: string }>();
+    const history = useHistory<TokenInfo>();
+
+    const [loading, setLoading] = useState(false);
+    const [errorMsg, setErrorMsg] = useState("");
+    const [editToken, setEditToken] = useState<PersonalAccessToken>();
+    const [token, setToken] = useState<EditPATData>({
+        name: "",
+        expirationDays: "30",
+        expirationDate: new Date(Date.now() + 30 * 24 * 60 * 60 * 1000),
```

Comment:
Ideally, this would be computed at the point of the request, not at the point of the view create